fix(alert): fix alert ui issues

ryanjdiercks opened the pull request:
## Jira

[UI Fixes for Alert Component](https://littlespoon.atlassian.net/browse/LS-4064)

## Test plan
- Review storybook: https://deploy-preview-686--littlespoon.netlify.app/?path=/story/design-system-alert--toast-stack
- Tests
- Desk check


## Screenshot
<img width="1196" alt="Screenshot 2022-11-07 at 22 58 11" src="https://user-images.githubusercontent.com/58109902/200424134-5fed5741-5b11-44f4-b4b0-7b7eb12db892.png">

<img width="969" alt="Screenshot 2022-11-07 at 22 58 27" src="https://user-images.githubusercontent.com/58109902/200424186-e7035f09-11ca-4f82-8066-310ecea12c74.png">

<img width="967" alt="Screenshot 2022-11-07 at 22 58 46" src="https://user-images.githubusercontent.com/58109902/200424237-c230bb3c-4a18-4439-bb7d-23565d8615f8.png">

huhgawz commented on packages/alert/src/AlertBase.tsx:
```diff
@@ -49,26 +50,41 @@ export const AlertWrapper = styled.div<BaseAlertProps>`
 
 export const AlertTitle = styled.span<Partial<AlertProps>>`
-  font: ${weight.bold} ${p3.fontSize} ${family};
-  line-height: ${p3.lineHeight};
+  font: ${weight.bold} ${p2.fontSize} ${family};
+  line-height: ${p2.lineHeight};
+
+  @media (max-width: ${breakpoints.sm}px) {
+    font: ${weight.bold} ${p3.fontSize} ${family};
+    line-height: ${p3.lineHeight};
+  }
```
Something like this could be done in JSX instead:
```jsx
<Typography variant={{'0': 'p2', [breakpoints.md]: 'p3'}}>Some text</Typography>
```

huhgawz commented on packages/alert/src/AlertBase.tsx:
```diff
@@ -49,26 +50,41 @@ export const AlertWrapper = styled.div<BaseAlertProps>`
 
 export const AlertDescription = styled.span<Partial<AlertProps>>`
-  font: ${weight.normal} ${p4.fontSize} ${family};
-  line-height: ${p4.lineHeight};
+  font: ${weight.normal} ${p3.fontSize} ${family};
+  line-height: ${p3.lineHeight};
   vertical-align: middle;
+
+  @media (max-width: ${breakpoints.sm}px) {
+    font: ${weight.normal} ${p4.fontSize} ${family};
+    line-height: ${p4.lineHeight};
+  }
 `
```
You could do something like this instead:
https://github.com/little-spoon-dev/little-spoon-frontend/blob/8b256db18696a31c36e2fe86b18a36f08776770d/app/pages/Account/MyBox/Content/ProductsGroups/ProductsGroup/styled.js#L81-L91

huhgawz commented on packages/alert/src/AlertBase.tsx:
```diff
@@ -49,26 +50,41 @@ export const AlertWrapper = styled.div<BaseAlertProps>`
 
 export const AlertActionLink = styled(Link)<Partial<AlertProps>>`
-  line-height: ${p4.lineHeight};
-  font: ${weight.bold} ${p4.fontSize} ${family};
+  line-height: ${p3.lineHeight};
+  font: ${weight.bold} ${p3.fontSize} ${family};
+
+  @media (max-width: ${breakpoints.sm}px) {
+    line-height: ${p4.lineHeight};
+    font: ${weight.bold} ${p4.fontSize} ${family};
+  }
 `
```
Something like this could be done in JSX instead:
```jsx
<Typography variant={{'0': 'p2', [breakpoints.md]: 'p3'}}>Some text</Typography>

huhgawz reviewed (changes requested):
Many requirements in [LS-4064](https://littlespoon.atlassian.net/browse/LS-4064) still need to be fixed:
<img width="439" src="https://user-images.githubusercontent.com/369338/198142952-1275fe45-1a91-4159-a319-cd423db24668.png">

ryanjdiercks commented:
> Some requirements in [LS-4064](https://littlespoon.atlassian.net/browse/LS-4064) still need to be fixed: <img alt="" width="439" src="https://user-images.githubusercontent.com/369338/198142952-1275fe45-1a91-4159-a319-cd423db24668.png">

I think I implemented these all there. 
- Developer can show/hide the close button with props.
- Implemented Toast Alert Stacking
- Increased the default Toast Alert display time to 6 secs.

and any reason why the Non Toast Alert shouldn't be animated when displayed? I couldn't find such spec in the document.

huhgawz commented:
> and any reason why the Non Toast Alert shouldn't be animated when displayed? I couldn't find such spec in the document.

The spec does not mention it but it feels weird, from a UX perspective, that a non toast alert is animated when displayed.  See the available component suites out there, in any of them a non toast alert is animated when displayed. For this reason I asked you to double check this behavior with Chantal

>     * Implemented Toast Alert Stacking

This image (taken from the [spec](https://zeroheight.com/3ddd0f892/p/76cbba-alert/b/70c440)) is clear about how stacking must behave:
<img width="669" alt="image" src="https://user-images.githubusercontent.com/369338/198726923-92be3907-08f2-4874-aed8-1918c3922c19.png">

huhgawz reviewed (changes requested):
In Storybook, Alerts with **Close** button are not being closed when the **Close** button is clicked, for example: https://deploy-preview-686--littlespoon.netlify.app/?path=/story/design-system-alert--alert-with-close. At least, generate a Storybook Action when the **Close** button is clicked

huhgawz commented:
> 



> > ```
> > * Implemented Toast Alert Stacking
> > ```
> 
> This image (taken from the [spec](https://zeroheight.com/3ddd0f892/p/76cbba-alert/b/70c440)) is clear about how stacking must behave:
> 
> <img alt="image" width="669" src="https://user-images.githubusercontent.com/369338/198726923-92be3907-08f2-4874-aed8-1918c3922c19.png">
> 
> > and any reason why the Non Toast Alert shouldn't be animated when displayed? I couldn't find such spec in the document.
> 
> The spec does not mention it but it feels weird, from a UX perspective, that a non toast alert is animated when displayed. In any of the existing component suites out there (for example [MUI Alert](https://mui.com/material-ui/react-alert/#basic-alerts)) a non toast alert is animated when displayed. For this reason, I asked you to double check this behavior with Chantal

Toast-Alerts are not bumped up yet

ryanjdiercks commented:
> In Storybook, Alerts with **Close** button are not being closed when the **Close** button is clicked, for example: https://deploy-preview-686--littlespoon.netlify.app/?path=/story/design-system-alert--alert-with-close. At least, generate a Storybook Action when the **Close** button is clicked

@huhgawz , could you help to implement it in storybook?
The close button function should be handled in parent component and I couldn't implement it in storybook.

huhgawz commented:
> > In Storybook, Alerts with **Close** button are not being closed when the **Close** button is clicked, for example: https://deploy-preview-686--littlespoon.netlify.app/?path=/story/design-system-alert--alert-with-close. At least, generate a Storybook Action when the **Close** button is clicked
> 
> @huhgawz , could you help to implement it in storybook? The close button function should be handled in parent component and I couldn't implement it in storybook.

@ryanjdiercks Take a look at https://github.com/little-spoon-dev/design-system/blob/main/packages/storybook/src/stories/Drawer.stories.tsx#L12

huhgawz commented on packages/alert/src/Alert.tsx:
```diff
@@ -129,29 +132,44 @@ export default function Alert({
+  }
+
+  if (!isAlertOpen) {
+    return <></>
```
```suggestion
    return null
```

JenniferBrody reviewed (approved):
QA done, design QA by Chantal too, approving
